### ISSUE TITLE
Add after-reload-settings hook

### ISSUE THREAD

**Issue by tbpassin** [leo-editor/leo-editor#3156]
Completed via PR #3208.

I've been thinking that a new hook would be useful, to signal that the settings have been reloaded.  Would that be feasible?

**Comment by edreamleo** [comment]
@tbpassin An `after-reload-settings` hook would be straightforward. `c.reloadConfigurableSettings` is the place to add this hook.

You should be aware of a small detail. At present `c.reloadConfigurableSettings` only calls `reloadSettings` methods. But there is an official, documented (I think) permissible alias: `reload_settings`.  So the lines:
```python
for obj in c.configurables:
    func = getattr(obj, 'reloadSettings', None)
```
probably should be:
```
for obj in c.configurables:
    func = (
        getattr(obj, 'reloadSettings', None)
        or getattr(obj, 'reload_settings', None)
    )
```
This change would be more resilient to future changes. Indeed, while researching this response I found that `ssm.reload_settings` had an `reloadSettings` alias, but `find.reload_settings` did not.  Rev 3d8981 in devel adds the missing alias, so the change above isn't strictly necessary *now*.

**Summary**

`c.reloadConfigurableSettings` looks like the place to add this hook.

**Comment by tbpassin** [comment]
Interesting... I didn't realize that the name reloadSettings was significant and useful.  I just thought it was a helpful mnemonic.

I wonder if this is documented somewhere.

**Comment by edreamleo** [comment]
@tbpassin You're welcome.

**Comment by edreamleo** [comment]
Closed via PR #3208.
